src/lib/validation.ts

```diff
@@ -154,6 +154,15 @@ export function isValidLabelsObject(obj: LabelObject): boolean {
 	});
 }
 
+export function isValidDeviceName(name: string): boolean {
+	// currently the only disallowed value in a device name is a newline
+	const newline = name.indexOf('\n') !== -1;
```

Comment:
This will also consider invalid a name such as `I\\nwant\\nnewlines` even though the new line is escaped and (I expect) shouldn't really cause issues. I don't suggest we should handle this edge-case though -- I see no real use-case -- just raising it.

Reply:
It won't match this, because `\\n` won't be converted to the newline character because of the escaping before it:
```
[10:06] ➜  ~ node
> a = "I\\nwant\\nnewlines"
'I\\nwant\\nnewlines'
> a.indexOf('\n')
-1